integration_test/integration_test.go

```diff
@@ -509,6 +509,140 @@ products:
 	assert.Equal(t, "dist/bar-0.1.0.sls.tgz\ndist/foo-0.1.0.sls.tgz\n", output)
 }
 
+func TestArtifactsBuildClean(t *testing.T) {
+	testProjectDir := setUpGödelTestAndDownload(t, testRootDir, gödelTGZ, version)
+	gittest.InitGitDir(t, testProjectDir)
+
+	distYml := `
+products:
+  foo:
+    build:
+      output-dir: ./foo/build
+      main-pkg: ./foo
+      os-archs:
+        - os: darwin
+          arch: amd64
+        - os: linux
+          arch: amd64
+  bar:
+    build:
+      main-pkg: ./bar
+      os-archs:
+        - os: linux
+          arch: amd64
+`
+	err := ioutil.WriteFile(path.Join(testProjectDir, "godel", "config", "dist.yml"), []byte(distYml), 0644)
+	require.NoError(t, err)
+
+	src := `package main
+	import "fmt"
+
+	func main() {
+		fmt.Println("hello, world!")
+	}`
+	err = os.MkdirAll(path.Join(testProjectDir, "foo"), 0755)
+	require.NoError(t, err)
+	err = ioutil.WriteFile(path.Join(testProjectDir, "foo", "foo.go"), []byte(src), 0644)
+	require.NoError(t, err)
+
+	err = os.MkdirAll(path.Join(testProjectDir, "bar"), 0755)
+	require.NoError(t, err)
+	err = ioutil.WriteFile(path.Join(testProjectDir, "bar", "bar.go"), []byte(src), 0644)
+	require.NoError(t, err)
+
+	gittest.CommitAllFiles(t, testProjectDir, "Commit files")
+	gittest.CreateGitTag(t, testProjectDir, "0.1.0")
+
+	execCommand(t, testProjectDir, "./godelw", "build")
+	fooBuild := execCommand(t, testProjectDir, "find", "foo/build", "-type", "f")
```

Comment:
Would prefer for tests to use standard Go library commands from `io` and `ioutils` package for doing things like checking for existence. The functionality provided by things like `ls` and `find` can be written directly in Go, and would prefer to do that whenever possible (would like to minimize the use of `execCommand` to do only the things that have to be shelled out).